### PR TITLE
feat(auth): add facade

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Removes the auth facade interface as it is not needed
- Adds the auth facade

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/auth/src/interfaces/auth-facade.interface.ts [hunk 1]
```diff
@@ -1,14 +0,0 @@
-import { Observable } from 'rxjs';
-import { Action } from '@ngrx/store';
-
-import { DaffStoreFacade } from '@daffodil/core';
-import { DaffAuthToken } from '../models/auth-token';
-
-export interface DaffAuthFacadeInterface<T extends DaffAuthToken> extends DaffStoreFacade<Action> {
```
Comment (damienwebdev) [inline libs/auth/src/interfaces/auth-facade.interface.ts:7]: Is this interface superseded by another? Why is it removed?

Comment (griest024) [reply libs/auth/src/interfaces/auth-facade.interface.ts:7]: Honestly this branch is so old I don't remember. Its not superseded. Should I add it back?

Comment (griest024) [reply libs/auth/src/interfaces/auth-facade.interface.ts:7]: It'll be needed for the testing facades, I'll add it back.

Comment (griest024) [reply libs/auth/src/interfaces/auth-facade.interface.ts:7]: added

**Pull request merged**